dataset/bucketby.cpp

```diff
@@ -208,16 +208,20 @@ DataArray bucketby_impl(const DataArrayConstView &array,
 
 DataArray bucketby(const DataArrayConstView &array,
                    const std::vector<VariableConstView> &edges) {
+  DataArrayConstView maybe_concat(array);
+  DataArray tmp;
   if (array.dtype() == dtype<bucket<DataArray>>) {
     // TODO The need for this check may be an indicator that we should support
     // adding another bucketed dimension via a separate function instead of
     // having this dual-purpose `bucketby`.
     for (const auto &edge : edges)
-      if (array.dims().contains(edge.dims().inner()))
-        throw std::runtime_error(
-            "Recursive buckets cannot be created with bucketby.");
+      if (array.dims().contains(edge.dims().inner())) {
+        // TODO Very inefficient if new edges extract only a small fraction
+        tmp = buckets::concatenate(maybe_concat, edge.dims().inner());
```

Comment:
Are you using this temporary for template deduction instead of calling `buckets::concatenate()` inside the constructor of `DataArrayView`?

Reply:
Can you reformulate the question?

Reply:
My question was that it seems you are not using `tmp` anywhere else, so why not simply
```
maybe_concat = DataArrayView(buckets::concatenate(maybe_concat, edge.dims().inner()));
```
?

Reply:
The `concatenate` will still still make a copy of (and concatenate) all the data, even if only 1% of the events is needed.

The reason for `tmp` is that it needs to keep `maybe_concat` alive, `DataArrayView` constructed over a rvalue `DataArray` would not work.